yt/frontends/gadget/io.py

```diff
@@ -67,7 +67,7 @@ def _read_particle_coords(self, chunks, ptf):
                 x = f["/%s/Coordinates" % ptype][si:ei, 0].astype("float64")
                 y = f["/%s/Coordinates" % ptype][si:ei, 1].astype("float64")
                 z = f["/%s/Coordinates" % ptype][si:ei, 2].astype("float64")
-                if ptype == self.ds._sph_ptype:
+                if ptype == self.ds._sph_ptypes[0]:
```

Comment:
I have to look further into the context. But for now, I'm wondering if the condition should be `ptype in self.ds._sph_ptypes` instead. I have the same question for other `ptype == self.ds._sph_ptypes[0]` occurrences.

Reply:
By the time you get down in here you will only see a `ptype` which is on-disk, so I think this logic is correct.

Reply:
That answered my question. Thanks!